cmd/crictl/info.go

```diff
@@ -25,36 +26,44 @@ import (
 	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
 )
 
-const (
-	criClientVersion = "v1alpha1"
-)
-
-var runtimeVersionCommand = cli.Command{
-	Name:  "info",
-	Usage: "Display runtime version information",
+var runtimeStatusCommand = cli.Command{
+	Name:      "info",
+	Usage:     "Display status and extra information of the container runtime",
+	ArgsUsage: "",
 	Action: func(context *cli.Context) error {
-		err := Version(runtimeClient, criClientVersion)
+		err := Info(runtimeClient)
 		if err != nil {
-			return fmt.Errorf("Getting the runtime version failed: %v", err)
+			return fmt.Errorf("Getting status of runtime failed: %v", err)
 		}
 		return nil
 	},
 	Before: getRuntimeClient,
 	After:  closeConnection,
 }
 
-// Version sends a VersionRequest to the server, and parses the returned VersionResponse.
-func Version(client pb.RuntimeServiceClient, version string) error {
-	request := &pb.VersionRequest{Version: version}
-	logrus.Debugf("VersionRequest: %v", request)
-	r, err := client.Version(context.Background(), request)
-	logrus.Debugf("VersionResponse: %v", r)
+// Info sends a StatusRequest to the server, and parses the returned StatusResponse.
+func Info(client pb.RuntimeServiceClient) error {
+	request := &pb.StatusRequest{}
+	logrus.Debugf("StatusRequest: %v", request)
+	r, err := client.Status(context.Background(), request)
+	logrus.Debugf("StatusResponse: %v", r)
 	if err != nil {
 		return err
 	}
-	fmt.Println("Version: ", r.Version)
-	fmt.Println("RuntimeName: ", r.RuntimeName)
-	fmt.Println("RuntimeVersion: ", r.RuntimeVersion)
-	fmt.Println("RuntimeApiVersion: ", r.RuntimeApiVersion)
-	return nil
+
+	statusInfo := struct {
+		Status   *pb.RuntimeStatus
+		ExtrInfo map[string]interface{}
+	}{
+		Status: r.Status,
+	}
+	for key, value := range r.Info {
+		var buf map[string]interface{}
+		err := json.Unmarshal([]byte(value), &buf)
+		if err != nil {
+			return err
+		}
+		statusInfo.ExtrInfo[key] = buf
```

Comment:
I guess in this way `buf` will be escaped.

Reply:
@Random-Liu  I have verified， it worked.

Reply:
This is my validation codes:
```go
package main

import "fmt"
import "encoding/json"

func main() {
	test := make(map[string]interface{})
	test["Int"] = 100
	test["String"] = "string"

	jsonMap := make(map[string]interface{})
	jsonMap["Num"] = 123
	jsonMap["Str"] = "abc"
	jsonByt, _ := json.Marshal(&jsonMap)
	test["JsonStr"]	= string(jsonByt)

	var jsonMap1 map[string]interface{}
	if err := json.Unmarshal(jsonByt, &jsonMap1); err == nil {
        	fmt.Println(jsonMap1)
	}

	test["JsonStr"] = jsonMap1
	byt, _ := json.Marshal(&test)
	fmt.Println("test-json-New:\n", string(byt))

	var testMap map[string]interface{}
	if err := json.Unmarshal(byt, &testMap); err == nil {
			fmt.Printf("%#v \n",testMap)
	}
}
```
output:
```bash
# go run main.go 
map[Num:123 Str:abc]
test-json-New:
 {"Int":100,"JsonStr":{"Num":123,"Str":"abc"},"String":"string"}
map[string]interface {}{"Int":100, "JsonStr":map[string]interface {}{"Num":123, "Str":"abc"}, "String":"string"}
```
But seems not strict, I will use your opinion. :)

Reply:
The value in map itself could be a json.